config/opal_config_pmix.m4

```diff
@@ -0,0 +1,200 @@
+    # an external that works.
+    external_pmix_happy=0
+    AS_IF([test "opal_pmix_mode" != "internal"],
+          [_OPAL_CONFIG_PMIX_EXTERNAL(
+              [external_pmix_happy=1
+               opal_pmix_mode="external"],
+              [AS_IF([test "$opal_pmix_mode" = "external"],
+                     [AC_MSG_ERROR([External PMIx requested but not found.])])])])
+
+    # external did not work out and customer did not specify external,
+    # so try the internal version.
+    AS_IF([test "$external_pmix_happy" = "0" -a "$internal_pmix_happy" = "1"],
+          [opal_pmix_mode="internal"
+           _OPAL_CONFIG_PMIX_INTERNAL_POST()])
+
+    AS_IF([test "$external_pmix_happy" = "0" -a "$internal_pmix_happy" = "0"],
+          [AC_MSG_ERROR([Could not find viable pmix build.])])
+
+    AC_SUBST(opal_pmix_CPPFLAGS)
+    AC_SUBST(opal_pmix_LDFLAGS)
+    AC_SUBST(opal_pmix_LIBS)
+
+    OPAL_SUMMARY_ADD([[Miscellaneous]], [[pmix]], [pmix], [$opal_pmix_mode])
+
+    OPAL_VAR_SCOPE_POP
+])
+
+dnl _OPAL_CONFIG_PMIX_EXTERNAL(action-if-happy, action-if-not-happy)
+dnl
+dnl only safe to call from OPAL_CONFIG_PMIX, assumes variables from
+dnl there are set.
+AC_DEFUN([_OPAL_CONFIG_PMIX_EXTERNAL], [
+    OPAL_VAR_SCOPE_PUSH([opal_pmix_CPPFLAGS_save opal_pmix_LDFLAGS_save opal_pmix_LIBS_save opal_pmix_external_support])
+
+    opal_pmix_CPPFLAGS_save=$CPPFLAGS
+    opal_pmix_LDFLAGS_save=$LDFLAGS
+    opal_pmix_LIBS_save=$LIBS
+
+    opal_pmix_external_support="yes"
+
+    AS_IF([test "$opal_libevent_mode" = "internal" -o "$opal_hwloc_mode" = "internal"],
+          [opal_pmix_external_support="no"
+           AS_IF([test "$opal_pmix_mode" = "external"],
+                 [AC_MSG_ERROR([Building against an external PMIx with an internal Libevent or HWLOC is unsupported.  Cannot continue.])])],
+          [AS_IF([test ! -z "$with_pmix_libdir"],
+                 [OPAL_CHECK_WITHDIR([pmix-libdir], [$with_pmix_libdir],
+                                     [libpmix.*])])
+
+           OPAL_CHECK_PACKAGE([opal_pmix],
+                              [pmix.h],
+                              [pmix],
+                              [PMIx_Init],
+                              [],
+                              [$with_pmix],
+                              [$with_pmix_libdir],
+                              [],
+                              [opal_pmix_external_support=no])
+
+           # need these set for the tests below.  If things fail, will undo at the end.
+           CPPFLAGS="$opal_pmix_CPPFLAGS_save $opal_pmix_CPPFLAGS"
+           LDFLAGS="$opal_pmix_LDFLAGS_save $opal_pmix_LDFLAGS"
+           LIBS="$opal_pmix_LIBS_save $opal_pmix_LIBS"
+
+           AS_IF([test "$opal_pmix_external_support" = "yes"],
+                 [AC_MSG_CHECKING([if external PMIx version is 3.0.0 or greater])
+                    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pmix_version.h>]],
+                          [[
+#if PMIX_VERSION_MAJOR < 3L
```

Comment:
I'm not sure OMPI (setting aside PRRTE) can support v3.0.0 - it may well require at least v3.1.5. I wouldn't hold this PR over it, but is something we should check prior to release.

Reply:
Good call out; I think testing is the right path.  3.0.0 was what the external component had been checking for, which is why I left it.